src/job/MomoJobBuilder.ts

```diff
@@ -0,0 +1,45 @@
+import { Handler, MomoJob } from './MomoJob';
+
+export class MomoJobBuilder implements Partial<MomoJob> {
+  name?: string;
+  interval?: string;
+  immediate?: boolean;
+  concurrency?: number;
+  maxRunning?: number;
+  handler?: Handler;
+
+  withName(name: string): this & Required<Pick<MomoJob, 'name'>> {
+    return Object.assign(this, { name: name });
+  }
+
+  withInterval(interval: string): this & Required<Pick<MomoJob, 'interval'>> {
+    return Object.assign(this, { interval: interval });
+  }
+
+  withImmediate(immediate: boolean): this & Pick<MomoJob, 'immediate'> {
+    return Object.assign(this, { immediate: immediate });
+  }
+
+  withConcurrency(concurrency: number): this & Pick<MomoJob, 'concurrency'> {
+    return Object.assign(this, { concurrency: concurrency });
+  }
+
+  withMaxRunning(maxRunning: number): this & Pick<MomoJob, 'maxRunning'> {
+    return Object.assign(this, { maxRunning: maxRunning });
+  }
+
+  withHandler(handler: Handler): this & Required<Pick<MomoJob, 'handler'>> {
+    return Object.assign(this, { handler: handler });
+  }
+
+  build(this: MomoJob): MomoJob {
```

Comment:
Don't we need some validation here? name, interval and handler are required for a MomoJob. The validation via the typesystem will not work for javascript.

Reply:
Good point! I've added the explicit validation.